main.py

```
@@ -38,21 +63,32 @@ def read_and_encode_file(file_path, encode=True):
     df['Date'] = pd.to_datetime(df['Date'], format='%m/%d/%Y')
     df.sort_values('Date', inplace=True, ascending=False)
+    
+    # Save as CSV
     df.to_csv(csv_path, index=False)
+    
+    # Save as pickle
+    with open(pickle_path, 'wb') as f:
+        pickle.dump(df, f)
 
     # 2. Merge all the years into one file
     file_names = sort_file_names(csv_dir)
```

Ruff (0.8.0)

`sort_file_names` may be undefined, or defined from star imports

(F405)